lineapy/data/types.py

```diff
@@ -405,13 +409,70 @@ class GlobalNode(BaseNode):
     call_id: LineaID
 
     def parents(self) -> Iterable[LineaID]:
+        yield from super().parents()
         yield self.call_id
 
 
+# class IfElseNode(BaseNode):
+#     """
+#     Represents an `if`/`else` block
+#     """
+
+#     node_type = Field(NodeType.IfElseNode, repr=False)
+
+#     # Points to the call node which forms the expression to test
+#     call_id: LineaID
+
+#     # LiteralNode containing the code of the block which does NOT get executed
+#     # Would be None if both the blocks get executed
+#     non_executed_contents = Optional[LineaID]
+
+#     def parents(self) -> Iterable[LineaID]:
+#         yield from super().parents()
+#         yield self.call_id
+#         if self.non_executed_contents:
+#             yield self.non_executed_contents
+
+
+class IfElseNode(BaseNode):
+    """
+    Represents the `if`, `else`, `elif` block
+    """
+
+    node_type = Field(NodeType.IfElseNode, repr=False)
+
+    # LiteralNode containing the code, for the block that is not executed
+    # If both `if` and `else` blocks are executed, this is None
+    unexec_contents: Optional[LineaID]
+
+    # Points to the call node which forms the expression to test
+    test_id: LineaID
+
+    # Points to a LiteralNode containing the `else` keyword if present
+    else_id: Optional[LineaID]
+
+    def parents(self) -> Iterable[LineaID]:
+        yield from super().parents()
+        yield self.test_id
+        if self.unexec_contents:
+            yield self.unexec_contents
+        if self.else_id:
+            yield self.else_id
+
+
+CNode = IfElseNode
```

Comment:
suggest changing to CNode = Type(IfElseNode)

Reply:
That will alter the semantics of `node_transformer.control_node()`, we need to return the object of type `ControlNode`, not an object of type `type`